src/BrazeKit-dev.js

```diff
@@ -66,32 +66,120 @@ var constructor = function () {
+                    parseProduct(_product)
+                );
+
+                var productName;
+                if (forwarderSettings.forwardSkuAsProductName === 'True') {
+                    productName = _product.Sku;
+                } else {
+                    productName = _product.Name;
+                }
+                product.name = getSanitizedValueForAppboy(productName);
+
+                eventAttributes.products.push(product);
+            });
+        }
+
+        kitLogger(
+            'appboy.logPurchase',
+            event.EventName,
+            event.ProductAction.TotalAmount,
+            event.CurrencyCode,
+            quantity,
+            eventAttributes
+        );
+
+        reportEvent = appboy.logPurchase(
+            event.EventName,
+            event.ProductAction.TotalAmount,
+            event.CurrencyCode,
+            quantity,
+            eventAttributes
+        );
+    }
+
+    function toUnderscore(string) {
+        return string
+            .split(/(?=[A-Z])/)
+            .join('_')
+            .toLowerCase();
+    }
+
+    function parseProduct(_product) {
+        var product = {};
+        for (var key in _product) {
+            switch (key) {
+                case 'Sku':
+                    product.id = _product.Sku;
+                    break;
+                case 'TotalAmount':
+                    product.total_product_amount = _product['TotalAmount'];
+                    break;
+                case 'Attributes':
+                    product.custom_attributes = _product.Attributes;
+                    break;
+                default:
+                    product[toUnderscore(key)] = _product[key];
+            }
+        }
+
+        return product;
+    }
+
+    function logPurchaseEventPerProduct(event) {
         if (event.ProductAction.ProductList) {
-            event.ProductAction.ProductList.forEach(function (product) {
+            event.ProductAction.ProductList.forEach(function(product) {
+                var productName;
+
+                if (forwarderSettings.forwardSkuAsProductName === 'True') {
+                    productName = product.Sku;
+                } else {
+                    productName = product.Name;
+                }
+
                 if (product.Attributes == null) {
                     product.Attributes = {};
                 }
                 product.Attributes['Sku'] = product.Sku;
 
-                var sanitizedProductName;
-                if (forwarderSettings.forwardSkuAsProductName === 'True') {
-                    sanitizedProductName = getSanitizedValueForAppboy(
-                        String(product.Sku)
-                    );
-                } else {
-                    sanitizedProductName = getSanitizedValueForAppboy(
-                        String(product.Name)
-                    );
-                }
+                var sanitizedProductName = getSanitizedValueForAppboy(
```

Comment:
Nitpick: I'd move this right after the end of the conditional `forwarderSettings.forwardSkuAsProductName` (line 165) since they both deal with `productName`